### PR TITLE
Resolve dependency issues between jar and zip maven publications

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Resolve dependency issues between jar and zip maven publications

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/3743
https://github.com/opensearch-project/custom-codecs/issues/38

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
Results:
```

% ls ~/.m2/repository/org/opensearch/opensearch-custom-codecs/2.10.0.0
opensearch-custom-codecs-2.10.0.0.jar  opensearch-custom-codecs-2.10.0.0-javadoc.jar  opensearch-custom-codecs-2.10.0.0.module  opensearch-custom-codecs-2.10.0.0.pom  opensearch-custom-codecs-2.10.0.0-sources.jar


% cat ~/.m2/repository/org/opensearch/opensearch-custom-codecs/2.10.0.0/opensearch-custom-codecs-2.10.0.0.pom | grep group
  <groupId>org.opensearch</groupId>
      <groupId>com.github.luben</groupId>
          <groupId>*</groupId>


% ls build/local-staging-repo/org/opensearch/opensearch-custom-codecs/2.10.0.0
opensearch-custom-codecs-2.10.0.0.jar         opensearch-custom-codecs-2.10.0.0-javadoc.jar         opensearch-custom-codecs-2.10.0.0.module         opensearch-custom-codecs-2.10.0.0.pom         opensearch-custom-codecs-2.10.0.0-sources.jar
opensearch-custom-codecs-2.10.0.0.jar.md5     opensearch-custom-codecs-2.10.0.0-javadoc.jar.md5     opensearch-custom-codecs-2.10.0.0.module.md5     opensearch-custom-codecs-2.10.0.0.pom.md5     opensearch-custom-codecs-2.10.0.0-sources.jar.md5
opensearch-custom-codecs-2.10.0.0.jar.sha1    opensearch-custom-codecs-2.10.0.0-javadoc.jar.sha1    opensearch-custom-codecs-2.10.0.0.module.sha1    opensearch-custom-codecs-2.10.0.0.pom.sha1    opensearch-custom-codecs-2.10.0.0-sources.jar.sha1
opensearch-custom-codecs-2.10.0.0.jar.sha256  opensearch-custom-codecs-2.10.0.0-javadoc.jar.sha256  opensearch-custom-codecs-2.10.0.0.module.sha256  opensearch-custom-codecs-2.10.0.0.pom.sha256  opensearch-custom-codecs-2.10.0.0-sources.jar.sha256
opensearch-custom-codecs-2.10.0.0.jar.sha512  opensearch-custom-codecs-2.10.0.0-javadoc.jar.sha512  opensearch-custom-codecs-2.10.0.0.module.sha512  opensearch-custom-codecs-2.10.0.0.pom.sha512  opensearch-custom-codecs-2.10.0.0-sources.jar.sha512


% cat build/local-staging-repo/org/opensearch/opensearch-custom-codecs/2.10.0.0/opensearch-custom-codecs-2.10.0.0.pom | grep group
  <groupId>org.opensearch</groupId>
      <groupId>com.github.luben</groupId>
          <groupId>*</groupId>

% ls build/local-staging-repo/org/opensearch/plugin/opensearch-custom-codecs/2.10.0.0/
opensearch-custom-codecs-2.10.0.0.pom      opensearch-custom-codecs-2.10.0.0.pom.sha1    opensearch-custom-codecs-2.10.0.0.pom.sha512  opensearch-custom-codecs-2.10.0.0.zip.md5   opensearch-custom-codecs-2.10.0.0.zip.sha256
opensearch-custom-codecs-2.10.0.0.pom.md5  opensearch-custom-codecs-2.10.0.0.pom.sha256  opensearch-custom-codecs-2.10.0.0.zip         opensearch-custom-codecs-2.10.0.0.zip.sha1  opensearch-custom-codecs-2.10.0.0.zip.sha512


% cat build/local-staging-repo/org/opensearch/plugin/opensearch-custom-codecs/2.10.0.0/opensearch-custom-codecs-2.10.0.0.pom | grep group
  <groupId>org.opensearch.plugin</groupId>
```

File: build.gradle [hunk 1]
```diff
@@ -68,33 +73,52 @@ dependencies {
...
+                    Node node = xml.asNode()
+                    node.appendNode('inceptionYear', '2021')
+
+                    Node license = node.appendNode('licenses').appendNode('license')
+                    license.appendNode('name', project.licenseName)
+                    license.appendNode('url', project.licenseUrl)
+
+                    Node developer = node.appendNode('developers').appendNode('developer')
+                    developer.appendNode('name', 'OpenSearch')
+                    developer.appendNode('url', 'https://github.com/opensearch-project/cusrom-codecs')
```
Comment (andrross) [inline build.gradle:104]:
```suggestion
                    developer.appendNode('url', 'https://github.com/opensearch-project/custom-codecs')
```

Comment (peterzhuamazon) [reply build.gradle]: Resolved, thanks.

**Review by andrross — APPROVED** [review]

**Review by andrross — COMMENTED** [review]
Is line 49 correct?

```
group 'org.opensearch'
```

**Comment by peterzhuamazon** [comment]
> Is line 49 correct?
> 
> ```
> group 'org.opensearch'
> ```

Yes, that is by design so group vs groupid is different here.

**Pull request merged**